### ISSUE TITLE
Update from 4.7.1 to 4.8.2 results in missing parts in the response

### ISSUE THREAD

**Issue by neonnoe** [strapi/strapi#16137]
### Required System information

- Node.js version: 17.9.1
- NPM version: 
- Strapi version: 4.8.2
- Database: affecting both local sqlite and live system with postgres
- Operating system:
- Is your project Javascript or Typescript: JavaScript

### Describe the bug

The same query shown in the screenshot results in different responses with missing parts with Strapi 4.8.2

<img width="1185" alt="image" src="https://user-images.githubusercontent.com/1913046/226205943-cbb41694-7700-4024-b3e6-bb30022ebdc3.png">

### Steps to reproduce the behavior

1. Create a dynamic pageZone
2. Add a content type with a relation to another content type to this page zone
3. In my case: Homepage has a dynamic zone containing a teaser content type, this teaser relates to an article
<img width="1046" alt="image" src="https://user-images.githubusercontent.com/1913046/226206153-52452ae8-c25c-41b8-8760-ccd0b8d774d7.png">
4. Query the homepage
5. With Strapi 4.8.2 the response is missing the article node in the JSON

### Expected behavior

As with the 4.7.1 version, the response should includes the article node. 
<img width="1056" alt="image" src="https://user-images.githubusercontent.com/1913046/226206361-acbf682d-b540-4879-86eb-fddf18c5a574.png">

### Code snippets

HTTP GET Query:
http://localhost:1337/api/home?populate[hero][populate]=*&populate[pageZone][populate]=*&locale=de

**Comment by Convly** [comment]
Hello and thank you for raising this issue.

In order to reproduce it, could you tell me a bit more about what kind of attribute is pageZone on your homepage? Is it a relation to a content type? A dynamic zone with its component?
If possible it would be awesome to have the declaration of the attributes for this specific chain of fields.

Thanks!

**Comment by neonnoe** [comment]
homepage is a single type, it includes the pageZone which is a Dynamic Zone. Within the Dynamic Zone there is an article teaser which is a component. The article teaser component has a reference to an article. The article is a collection type. I hope this helps to reproduce the scenario.

**Comment by Convly** [comment]
Thanks a lot, I'll try to reproduce it and keep you updated.

**Comment by Convly** [comment]
It seems like it's working on my side, do you have customized endpoints for your homepage? Do you have the right permissions set?

The data structure
![image](https://user-images.githubusercontent.com/25851739/226579009-feedc25b-dfa9-4246-a2ad-daf491fa6a33.png)
The API results
![image](https://user-images.githubusercontent.com/25851739/226578782-2089a91a-2385-4332-9966-db2575fc6e34.png)
The config
![image](https://user-images.githubusercontent.com/25851739/226579204-892dbeb7-825e-4fc6-a2c3-6791847fc23e.png)

**Comment by neonnoe** [comment]
I found the reason why the info was missing in the response. By accident, I set the field to private.

<img width="591" alt="image" src="https://user-images.githubusercontent.com/1913046/226744096-d03c53e8-325a-4938-a786-689763a1bcf8.png">

It seems that earlier versions of strapi ignored this property and included the field in the response. With the latest version, the data will be ignored when it has the private property.
